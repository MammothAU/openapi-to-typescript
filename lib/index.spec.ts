import { GenerateTypes } from './index'

describe('openapi-typescript', () => {
  describe('GenerateTypes', () => {
    describe('given an empty openapi schema', () => {
      const schema = {
        components: {
          schemas: {}
        },
        paths: {}
      }
      it('should respond nothing', async () => {
        const typings = await GenerateTypes(schema)
        expect(typings).toMatch(/^\s*$/)
      })
    })

    describe('given the Petstore schema', () => {
      const schema = require('../fixtures/petstore.json')

      it('should output types for components.schemas ', async () => {
        const typings = await GenerateTypes(schema)
        expect(typings).toContain('export interface Pet')
        expect(typings).toContain('export type Pets = (Pet)[]')
      })
    })

    describe('given the Petstore Expanded schema', () => {
      const schema = require('../fixtures/petstoreExpanded.json')

      it('should output types for components.schemas ', async () => {
        const typings = await GenerateTypes(schema)
        expect(typings).toContain('export type Pet = NewPet & {')
        // expect(typings).toMatch(/^\s*$/)
      })
    })

  })
})