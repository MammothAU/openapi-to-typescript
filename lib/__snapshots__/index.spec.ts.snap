// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateTypings given an empty openapi schema should match clientStore snapshot 1`] = `""`;

exports[`GenerateTypings given an empty openapi schema should match typeStore snapshot 1`] = `""`;

exports[`GenerateTypings given the Petstore Expanded schema FetchClientFormatter should match clientStore snapshot 1`] = `
"const fetch = require('node-fetch')
const pick = (obj:any, keys:string[]) => keys.reduce((picked, key) => obj[key] !== undefined ? Object.assign(picked, {[key]: obj[key]}) : picked, {})
const encodeQuery = (obj:any, keys:string[]) => require('querystring').encode(pick(obj, keys))
const API_URL = \\"http://petstore.swagger.io/api\\"
const FindPets =
      async (body:FindPetsRequest, options:any):Promise<FindPetsResult> => {
        return fetch(API_URL, {
          ...options,
          method: \\"get\\",
          body: JSON.stringify(body)
        }).then((res:any) => res.json())
      }
const AddPet =
      async (body:AddPetRequest, options:any):Promise<AddPetResult> => {
        return fetch([API_URL, encodeQuery(body, [])].filter(x=>x).join('?'), {
          ...options,
          method: \\"post\\",
          body: JSON.stringify(body)
        }).then((res:any) => res.json())
      }
const FindPetById =
      async (body:FindPetByIdRequest, options:any):Promise<FindPetByIdResult> => {
        return fetch([API_URL, encodeQuery(body, [])].filter(x=>x).join('?'), {
          ...options,
          method: \\"get\\",
          body: JSON.stringify(body)
        }).then((res:any) => res.json())
      }
const DeletePet =
      async (body:DeletePetRequest, options:any):Promise<DeletePetResult> => {
        return fetch([API_URL, encodeQuery(body, [])].filter(x=>x).join('?'), {
          ...options,
          method: \\"delete\\",
          body: JSON.stringify(body)
        }).then((res:any) => res.json())
      }"
`;

exports[`GenerateTypings given the Petstore Expanded schema FetchClientFormatter should match typeStore snapshot 1`] = `
"export type Pet = NewPet & {
  id: number;
  [k: string]: any;
};

export interface NewPet {
  name: string;
  tag?: string;
  [k: string]: any;
}

export interface Error {
  code: number;
  message: string;
  [k: string]: any;
}

export interface FindPetsRequest {
  tags: string[]
  limit: number
}
export type FindPetsResult = (Pet)[];

export type FindPetsFallback = Error;

export type FindPets = (payload: FindPetsRequest) => Promise<FindPetsResult>;
export type AddPetRequest = NewPet
export type AddPetResult = Pet;

export type AddPetFallback = Error;

export type AddPet = (payload: AddPetRequest) => Promise<AddPetResult>;
export interface FindPetByIdRequest {
  id: number
}
export type FindPetByIdResult = Pet;

export type FindPetByIdFallback = Error;

export type FindPetById = (payload: FindPetByIdRequest) => Promise<FindPetByIdResult>;
export interface DeletePetRequest {
  id: number
}
export type DeletePetResult = any
export type DeletePetFallback = Error;

export type DeletePet = (payload: DeletePetRequest) => Promise<DeletePetResult>;"
`;

exports[`GenerateTypings given the Petstore Expanded schema should match clientStore snapshot 1`] = `""`;

exports[`GenerateTypings given the Petstore Expanded schema should match typeStore snapshot 1`] = `
"export type Pet = NewPet & {
  id: number;
  [k: string]: any;
};

export interface NewPet {
  name: string;
  tag?: string;
  [k: string]: any;
}

export interface Error {
  code: number;
  message: string;
  [k: string]: any;
}

export interface FindPetsRequest {
  tags: string[]
  limit: number
}
export type FindPetsResult = (Pet)[];

export type FindPetsFallback = Error;

export type AddPetRequest = NewPet
export type AddPetResult = Pet;

export type AddPetFallback = Error;

export interface FindPetByIdRequest {
  id: number
}
export type FindPetByIdResult = Pet;

export type FindPetByIdFallback = Error;

export interface DeletePetRequest {
  id: number
}
export type DeletePetResult = any
export type DeletePetFallback = Error;
"
`;

exports[`GenerateTypings given the Petstore schema should match clientStore snapshot 1`] = `""`;

exports[`GenerateTypings given the Petstore schema should match typeStore snapshot 1`] = `
"export interface Pet {
  id: number;
  name: string;
  tag?: string;
  [k: string]: any;
}

export type Pets = (Pet)[];

export interface Error {
  code: number;
  message: string;
  [k: string]: any;
}

export interface ListPetsRequest {
  limit: number
}
export type ListPetsResult = Pets;

export type ListPetsFallback = Error;

export interface CreatePetsRequest {
}
export type CreatePetsResult = any
export type CreatePetsFallback = Error;

export interface ShowPetByIdRequest {
  petId: string
}
export type ShowPetByIdResult = Pets;

export type ShowPetById204 = any
export type ShowPetByIdFallback = Error;
"
`;
